fiber/chain_interactions/commitments.py

```diff
@@ -0,0 +1,224 @@
+    if not field:
+        return {str(None): b''}
+
+    data_type, data = field
+
+    if data_type == DataFieldType.RAW:
+        serialized_data_type = DataFieldType.RAW.value + str(len(data))
+    else:
+        serialized_data_type = data_type.value
+
+    return {serialized_data_type: data}
+
+
+def _deserialize_field(field: dict[str, bytes | str]) -> CommitmentDataField:
+    data_type, data = field.items().__iter__().__next__()
+
+    if data_type == str(None):
+        return None
+
+    if data_type.startswith(DataFieldType.RAW.value):
+        expected_length = int(data_type[len(DataFieldType.RAW.value):])
+        data_type = DataFieldType.RAW.value
+        data = bytes.fromhex(data[2:])
+
+        if len(data) != expected_length:
+            raise ValueError(f"Got commitment raw field expecting {expected_length} data but got {len(data)} data")
+
+    return DataFieldType(data_type), data
+
+
+@retry(
+    stop=stop_after_attempt(3),
+    wait=wait_exponential(multiplier=1, min=1, max=4),
+    reraise=True,
+)
+def _query_commitment(
+    substrate: SubstrateInterface,
+    netuid: int,
+    hotkey: str,
+    block: int | None = None,
+) -> ScaleType:
+    return substrate.query(
+        module="Commitments",
+        storage_function="CommitmentOf",
+        params=[netuid, hotkey],
+        block_hash=(None if block is None else substrate.get_block_hash(block)),  # type: ignore
+    )
+
+
+@retry(
+    stop=stop_after_attempt(3),
+    wait=wait_exponential(multiplier=1.5, min=2, max=5),
+    reraise=True,
+)
+def set_commitment(
+    substrate_interface: SubstrateInterface,
+    keypair: Keypair,
+    netuid: int,
+    fields: list[CommitmentDataField],
+    wait_for_inclusion: bool = False,
+    wait_for_finalization: bool = False,
+) -> tuple[bool, str | None]:
+    """
+    Commit custom fields to the chain
+    Arguments:
+        fields: A list of fields as data type to value tuples, for example (DataFieldType.RAW, b'hello world')
+    """
+
+    mapped_fields = [[
+        _serialize_field(field)
+        for field in fields
+    ]]
+
+    call = substrate_interface.compose_call(
+        call_module="Commitments",
+        call_function="set_commitment",
+        call_params={
+            "netuid": netuid,
+            "info": {
+                "fields": mapped_fields,
+            },
+        },
+    )
+
+    extrinsic_to_send = substrate_interface.create_signed_extrinsic(call=call, keypair=keypair)
+
+    response = substrate_interface.submit_extrinsic(
+        extrinsic_to_send,
+        wait_for_inclusion=wait_for_inclusion,
+        wait_for_finalization=wait_for_finalization,
+    )
+
+    if not wait_for_finalization and not wait_for_inclusion:
+        return True, "Not waiting for finalization or inclusion."
+
+    response.process_events()
+
+    if response.is_success:
+        return True, "Successfully submitted commitment."
+
+    return False, format_error_message(response.error_message)
```

Comment:
What's ever done with this error messaeg? can we just logger.error() it instead, and then just return a bool? makes life easier